_posts/2024-06-12-branching-and-versioning.adoc

```diff
@@ -0,0 +1,84 @@
+---
+layout: post
+title: 'Why you should always update to the latest micro versions!'
+date: 2024-06-12
+tags: branching versioning
+synopsis: 'How does quarkus use Git and the implication on releases and Quarkus versions'
+author: geoand
+---
+:imagesdir: /assets/images/posts/branching-and-versioning
+
+We often get questions about how the Quarkus maintainers handle multiple branches and how this impacts releases. This short post attempts to provide the necessary information for answering such questions.
+
+== Development branch
+
+When a maintainer or contributor wants to make a change to Quarkus source code, a Pull Request is opened against the `main` branch.
+This means that all changes once merged are part of the `main` branch and there are no feature branches (a practice that is sometimes referred to as trunk based development).
+
+NOTE: The Maven version used on the `main` branch is always `999-SNAPSHOT`
```

Comment:
its actually used on *all* branches which actually is problematic imo. we should be using i.e. 3.11.9999-SNAPSHOT or something similar.

not sure how many realize that when switching to maint. branches they actually should nuke all 999-SNAPSHOT in ~/.m2 to not risk pollution...

Reply:
> its actually used on all branches

I don't think that's the case anymore

Reply:
Right, but for the sake of this blog post it does not really matter